internal/controller/services/monitoring/resources/monitoring-stack.tmpl.yaml

```diff
@@ -13,7 +13,9 @@ spec:
       resources:
         requests:
           storage: {{.StorageSize}}
+{{- if .Replicas }}
     replicas: {{.Replicas}}
+{{- end }}
```

Comment:
If is not needed because this template will be processed only if metrics are defined, which means replicas will be defined too (either by user or our defaulting automation).